pypi/build.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+set -e -o pipefail
+
+mkdir -p /build
+cd /build
+
+echo "Hello from PyPi build.sh"
+
+BUILD_DIR=`pwd`
+N_CORES=`nproc`
+
+cp -r /boolector .
+
+# Setup dependencies
+cd boolector
+/bin/sh ./contrib/setup-btor2tools.sh
+/bin/sh ./contrib/setup-cadical.sh
+/bin/sh ./contrib/setup-lingeling.sh
+
+#********************************************************************
+#* boolector
+#********************************************************************
+cd ${BUILD_DIR}
+
+cd boolector
+
+./configure.sh --shared --prefix /usr
+
+cd build
+
+make -j${N_CORES}
+
+make install
+
+#********************************************************************
+#* pyboolector
+#********************************************************************
+cd ${BUILD_DIR}
+rm -rf pyboolector
+
+# Specify path to CmakeLists.txt so setup.py can extract the version
+export CMAKELISTS_TXT=/boolector/CMakeLists.txt
+
+cp -r /boolector/pypi pyboolector
+
+# Grab the main license file
+cp /boolector/COPYING pyboolector/LICENSE
+
+cd pyboolector
+
+for py in cp35-cp35m cp36-cp36m cp37-cp37m cp38-cp38 cp39-cp39; do
+  echo "Python: ${py}"
+  python=/opt/python/${py}/bin/python
+  cd ${BUILD_DIR}/pyboolector
+  rm -rf src
+  cp -r ${BUILD_DIR}/boolector/src/api/python src
+  sed -i -e 's/override//g' \
+     -e 's/noexcept/_GLIBCXX_USE_NOEXCEPT/g' \
+     -e 's/\(BoolectorException (const.*\)/\1\n    virtual ~BoolectorException() _GLIBCXX_USE_NOEXCEPT {}/' \
+       src/pyboolector_abort.cpp
+  mkdir -p src/utils
+  cp ${BUILD_DIR}/boolector/src/*.h src
+  cp ${BUILD_DIR}/boolector/src/utils/*.h src/utils
+  $python ./src/mkenums.py ./src/btortypes.h ./src/pyboolector_enums.pxd
+  $python setup.py sdist bdist_wheel
+done
```

Comment:
Just out of curiosity: Why can't the library produced by `./configure --python` be used here?

Reply:
The only way I could find to produce a 'wheel' (Python distribution package) that PyPi would accept was to use the Python build commands via setup.py. I think part of this is due to the package auditing that exists to ensure that packages posted on PyPi will truly be compliant with the platform specifications they purport to be compatible with.

Reply:
BTW, one option going forward (thinking Bitwuzla) would be to build all Python packages via a setup.py. The CMakeLists.txt would be slightly different, and would invoke Python on the setup.py instead of directly building the shared library. The advantage would be a single way of building the Python package.

Reply:
Oh ok. Do you have experience with scikit-build? Look like this is allows better integration with a cmake build system.

Reply:
Unfortunately, I don't. I think I looked at scikit-build early in my process of learning about building Python extensions, and think my conclusion was that it was optimized for certain types of extension but lacked flexibility. Certainly not in a position to offer a knowledgeable opinion, though.

I've been working on building Python extensions as part of the regular build in the context of a different project. Thus far, it doesn't look too difficult. Here's the CMakeList fragment that invokes the actual build:
https://github.com/PSSTools/pssparser/blob/67fdb8d0cd41874f80f6e8c1acb60a83ac6cc156/ext/CMakeLists.txt#L7-L20

This, in turn, is driven by a script run inside the 'manylinux2010' Docker image to produce all the Python-version variants:
https://github.com/PSSTools/pssparser/blob/67fdb8d0cd41874f80f6e8c1acb60a83ac6cc156/.github/workflows/pypi.sh#L26-L38

Fortunately, Makefile dependencies cause only the Python extension (and not all the other code) to be rebuilt for each Python version